crates/biome_service/src/workspace/server.rs

```diff
@@ -1227,54 +1227,55 @@ impl Workspace for WorkspaceServer {
         let parse = self.get_parse(&path)?;
         let language = self.get_file_source(&path);
         let capabilities = self.features.get_capabilities(language);
-        let (diagnostics, errors, skipped_diagnostics) =
-            if let Some(lint) = capabilities.analyzer.lint {
-                let settings = self
-                    .projects
-                    .get_settings_based_on_path(project_key, &path)
-                    .ok_or_else(WorkspaceError::no_project)?;
-
-                let plugins = self
-                    .get_analyzer_plugins_for_project(
-                        settings.source_path().unwrap_or_default().as_path(),
-                        &settings.get_plugins_for_path(&path),
-                    )
-                    .map_err(WorkspaceError::plugin_errors)?;
-
-                let results = lint(LintParams {
-                    parse,
-                    settings: &settings,
-                    path: &path,
-                    only,
-                    skip,
-                    language,
-                    categories,
-                    module_graph: self.module_graph.clone(),
-                    project_layout: self.project_layout.clone(),
-                    suppression_reason: None,
-                    enabled_rules,
-                    pull_code_actions,
-                    plugins: if categories.contains(RuleCategory::Lint) {
-                        plugins
-                    } else {
-                        Vec::new()
-                    },
-                });
+        let (diagnostics, errors, skipped_diagnostics) = if categories.is_lint()
```

Comment:
`format_with_guard()` calls `Workspace::pull_diagnostics` to collect parse diagnostics, but it also lead calling `lint()`, which is not necessary and takes some time.